src/crimson/os/seastore/async_cleaner.cc

```diff
@@ -1057,6 +1057,17 @@ SegmentCleaner::do_reclaim_space(
     std::size_t &reclaimed,
     std::size_t &runs)
 {
+  // Extents satisfying any of the following requirements
+  // are considered DEAD:
+  // 1. can't find the corresponding mapping in both the
+  // 	backref tree and the backref cache;
+  // 2. the extent is logical and the lba tree doesn't
+  // 	have a mapping for the laddr that the extent
+  // 	corresponds to;
+  // 3. the extent is physical and doesn't exist in the
+  // 	lba tree;
+  // 4. the extent is logical and the mapping in the lba
+  // 	tree points to another paddr.
```

Comment:
Seems it can be reduced to 2 major cases:
* The extent is logical, but its lba mapping doesn't exist or match in the lba tree.
* The extent is physical, but it doesn't exist in the lba tree, backref tree or backref cache.

Reply:
Just updated, please take a look again.

Reply:
Out of curious, why split 1 and 3, or what's important that needs to make 1 and 3 different?

Reply:
Because logical extents are also retrieved from the backref tree/cache, and then the lba tree queries are conducted to confirm that they are still valid.

Reply:
OK, it seems to be identical to that the logical extent's lba mapping doesn't exist, given the more fundamental invariant that the lba tree and backref tree must match each other.